spec/controllers/chapters_controller_spec.rb

```diff
@@ -4,74 +4,554 @@
+        expect { post :create, work_id: work.id, chapter: @chapter_attributes }.to change(Chapter, :count)
+        expect(work.chapters.count).to eq 2
+      end
+
+      it "updates the works wip length when given" do
+        @chapter_attributes[:wip_length] = 3
+        expect(work.wip_length).to eq 1
+        post :create, work_id: work.id, chapter: @chapter_attributes
+        expect(assigns[:work].wip_length).to eq 3
+      end
+
+      context "when chapter has invalid pseuds" do
+        before do
+          allow_any_instance_of(Chapter).to receive(:invalid_pseuds).and_return([user.pseuds.first])
+        end
+        it "renders choose coauthor if chapter if valid" do
+          post :create, work_id: work.id, chapter: @chapter_attributes
+          expect(response).to render_template("_choose_coauthor")
+        end
+
+        it "renders new if chapter is not valid" do
+          post :create, work_id: work.id, chapter: { content: "" }
+          expect(response).to render_template(:new)
+        end
+      end
+
+      context "when chapter has ambiguous pseuds" do
+        before do
+          allow_any_instance_of(Chapter).to receive(:ambiguous_pseuds).and_return([user.pseuds.first])
+        end
+        it "renders choose coauthor if chapter if valid" do
+          post :create, work_id: work.id, chapter: @chapter_attributes
+          expect(response).to render_template("_choose_coauthor")
+        end
+
+        it "renders new if chapter is not valid" do
+          post :create, work_id: work.id, chapter: { content: "" }
+          expect(response).to render_template(:new)
+        end
+      end
+
+      it "renders new if the edit button has been clicked" do
+        post :create, work_id: work.id, chapter: @chapter_attributes, edit_button: true
+        expect(response).to render_template(:new)
+      end
+
+      it "redirects if the cancel button has been clicked" do
+        post :create, work_id: work.id, chapter: @chapter_attributes, cancel_button: true
+        expect(response).to have_http_status :redirect
+      end
+
+      it "updates the work's major version" do
+        expect(work.major_version).to eq(1)
+        post :create, work_id: work.id, chapter: @chapter_attributes
+        expect(assigns[:work].major_version).to eq(2)
+      end
+
+      context "when the post without preview button is clicked" do
+        context "when the chapter and work are valid" do
+          it "posts the chapter" do
+            post :create, work_id: work.id, chapter: @chapter_attributes, post_without_preview_button: true
+            expect(assigns[:chapter].posted).to be true
+          end
+
+          it "posts the work if the work was not posted before" do
+            @unposted_work = create(:work, authors: [user.pseuds.first])
+            post :create, work_id: @unposted_work.id, chapter: @chapter_attributes, post_without_preview_button: true
+            expect(assigns[:work].posted).to be true
```

Comment:
Oh, okay! I was expecting the success message to be different here ("Work successfully posted, it may take a while to appear in listings," etc), since this test uses an unposted work and the next test uses a posted work. However, I see now that's not the current behavior.